test/notifications/display_test.dart

```diff
@@ -151,53 +181,87 @@ void main() {
               ..summaryText.equals(data.realmUri.toString()))
             ..autoCancel.equals(true)
-            ..contentIntent.isNull()
+            ..contentIntent.isNull(),
         ]);
+      }
+
+      check(testBinding.androidNotificationHost.takeNotifyCalls())
+        .deepEquals(notifyCallsChecks);
+      testBinding.androidNotificationHost.clearActiveNotifications();
     }
 
-    Future<void> checkNotifications(FakeAsync async, MessageFcmMessage data, {
+    Future<void> checkNotifications(FakeAsync async, List<MessageFcmMessage> messages, {
       required String expectedTitle,
       required String expectedTagComponent,
+      required bool expectedGroup,
     }) async {
       // We could just call `NotificationDisplayManager.onFcmMessage`.
       // But this way is cheap, and it provides our test coverage of
       // the logic in `NotificationService` that listens for these FCM messages.
 
-      testBinding.firebaseMessaging.onMessage.add(
-        RemoteMessage(data: data.toJson()));
-      async.flushMicrotasks();
-      checkNotification(data, expectedTitle: expectedTitle,
+      for (final data in messages) {
+        testBinding.firebaseMessaging.onMessage.add(
+          RemoteMessage(data: data.toJson()));
+        async.flushMicrotasks();
+      }
+      checkNotification(messages,
+        expectedGroup: expectedGroup,
+        expectedTitle: expectedTitle,
         expectedTagComponent: expectedTagComponent);
 
-      testBinding.firebaseMessaging.onBackgroundMessage.add(
-        RemoteMessage(data: data.toJson()));
-      async.flushMicrotasks();
-      checkNotification(data, expectedTitle: expectedTitle,
+      for (final data in messages) {
+        testBinding.firebaseMessaging.onBackgroundMessage.add(
+          RemoteMessage(data: data.toJson()));
+        async.flushMicrotasks();
+      }
+      checkNotification(messages,
+        expectedGroup: expectedGroup,
+        expectedTitle: expectedTitle,
         expectedTagComponent: expectedTagComponent);
     }
 
     test('stream message', () => awaitFakeAsync((async) async {
       await init();
       final stream = eg.stream();
       final message = eg.streamMessage(stream: stream);
-      await checkNotifications(async, messageFcmMessage(message, streamName: stream.name),
+      await checkNotifications(async, [messageFcmMessage(message, streamName: stream.name)],
+        expectedGroup: true,
         expectedTitle: '#${stream.name} > ${message.topic}',
         expectedTagComponent: 'stream:${message.streamId}:${message.topic}');
     }));
 
+    test('multiple stream messages', () => awaitFakeAsync((async) async {
+      await init();
+      final stream = eg.stream(streamId: 1, name: 'stream 1');
+      final message1 = eg.streamMessage(id: 101, stream: stream, timestamp: 1);
+      final message2 = eg.streamMessage(id: 102, stream: stream, timestamp: 2);
+      final message3 = eg.streamMessage(id: 103, stream: stream, timestamp: 3);
```

Comment:
This test is relying on all three of these being in the same conversation. So it should make that explicit in the data, by specifying the topic.

OTOH I don't think there's a need to specify the message IDs. See the `eg.streamMessage` doc. If we don't need to specify them, then best to leave them out, because they're totally boring as far as this test is concerned.

Reply:
Similarly the stream ID and name to `eg.stream`.